sonar-platform-plugin/src/main/java/io/ballerina/sonar/SonarPlatformPlugin.java

```diff
@@ -0,0 +1,158 @@
+ *  * under the License.
+ *
+ */
+
+package io.ballerina.sonar;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import io.ballerina.scan.Issue;
+import io.ballerina.scan.PlatformPluginContext;
+import io.ballerina.scan.StaticCodeAnalysisPlatformPlugin;
+import io.ballerina.scan.internal.IssueImpl;
+import org.apache.commons.lang3.SystemUtils;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.ballerina.sonar.Constants.RULE_PREFIX;
+import static io.ballerina.sonar.Constants.END_LINE;
+import static io.ballerina.sonar.Constants.END_LINE_OFFSET;
+import static io.ballerina.sonar.Constants.FILE_NAME;
+import static io.ballerina.sonar.Constants.FILE_PATH;
+import static io.ballerina.sonar.Constants.FORWARD_SLASH;
+import static io.ballerina.sonar.Constants.ISSUE_FILE_PATH;
+import static io.ballerina.sonar.Constants.MESSAGE;
+import static io.ballerina.sonar.Constants.RULE_ID;
+import static io.ballerina.sonar.Constants.RULE_KIND;
+import static io.ballerina.sonar.Constants.SOURCE;
+import static io.ballerina.sonar.Constants.START_LINE;
+import static io.ballerina.sonar.Constants.PLATFORM_NAME;
+import static io.ballerina.sonar.Constants.START_LINE_OFFSET;
+
+public class SonarPlatformPlugin implements StaticCodeAnalysisPlatformPlugin {
+    private PlatformPluginContext platformPluginContext;
+    private final List<String> processBuilderArguments = new ArrayList<>();
+    private final ProcessBuilder processBuilder = new ProcessBuilder();
+    private final PrintStream outputStream = System.out;
+
+    @Override
+    public String platform() {
+        return PLATFORM_NAME;
+    }
+
+    @Override
+    public void init(PlatformPluginContext platformPluginContext) {
+        this.platformPluginContext = platformPluginContext;
+        if (SystemUtils.IS_OS_WINDOWS) {
+            processBuilderArguments.add("cmd");
+            processBuilderArguments.add("/c");
+        } else {
+            processBuilderArguments.add("sh");
+            processBuilderArguments.add("-c");
+        }
+        processBuilderArguments.add("sonar-scanner");
+        processBuilderArguments.add("-Dsonar.exclusions=" +
+                "'" +
+                "**/*.java," +
+                "**/*.xml," +
+                "**/*.yaml," +
+                "**/*.go," +
+                "**/*.kt," +
+                "**/*.js," +
+                "**/*.html," +
+                "**/*.YAML" +
+                ",**/*.rb," +
+                "**/*.scala," +
+                "**/*.py" +
+                "'");
+    }
+
+    @Override
+    public void onScan(List<Issue> issues) {
+        saveIssues(ISSUE_FILE_PATH, issues);
+        if (!platformPluginContext.initiatedByPlatform()) {
```

Comment:
Check the inverse and return early. Reduces indentation for the rest of the code. Please check and incorporate this pattern wherever applicable.